src/DefaultManifold.jl

```diff
@@ -9,11 +9,20 @@ This manifold further illustrates how to type your manifold points and tangent v
 that the interface does not require this, but it might be handy in debugging and educative
 situations to verify correctness of involved variabes.
 """
-struct DefaultManifold{𝔽,T<:NTuple{N,Int} where {N}} <: AbstractManifold{𝔽}
+struct DefaultManifold{𝔽,T<:AbstractManifoldSize} <: AbstractManifold{𝔽}
     size::T
 end
-function DefaultManifold(n::Vararg{Int}; field = ℝ)
-    return DefaultManifold{field,typeof(n)}(n)
+function ManifoldsBase.DefaultManifold(
+    n::Vararg{Int};
+    field = ManifoldsBase.ℝ,
+    static = false,
```

Comment:
We could maybe – for consistency, do this as a keyword? It will be a keyword for Euclidean anyways.

Also just `static` might not be so informative as a keyword, maybe we can find a nicer name?

Reply:
This is a keyword now (notice the semicolon).

Other possible names would be perhaps `static_size`, `specialize_for_size`, `specialize_size` or something like that? Nothing sounds particularly convincing.

Reply:
Hm, then maybe we should also use `TypeParameter` and `FieldParameter` names for the types?

Reply:
That sounds good :)